### PR TITLE
Feature/ReSwift4.x

### PR DESCRIPTION
**Pull request by rpassis**
As per issue #64, with the breaking changes in ReSwift 4.x the router needed to be updated with a corresponding signature. This pull request incorporates all these changes, including updated Cartfile and unit tests.

### CODE REVIEW

**Comment by lkzhao** [comment]
@rpassis Should probably change the podspec file to point to ReSwift 4.0 too.

File: ReSwiftRouterTests/ReSwiftRouterIntegrationTests.swift [hunk 1]
```diff
@@ -107,7 +103,7 @@ class SwiftFlowRouterIntegrationTests: QuickSpec {
 
                     let routable = FakeRootRoutable()
                     let _ = Router(store: store, rootRoutable: routable) { state in
-                        state.navigationState
+                        state.select({$0.navigationState})
```
Comment (Ben-G) [inline ReSwiftRouterTests/ReSwiftRouterIntegrationTests.swift]: How about using trailing closure syntax here and in other cases below?

E.g.:

```
state.select { $0.navigationState }
```

**Comment by Ben-G** [comment]
Thanks a lot @rpassis. Only one minor comment. Other than that this LGTM!

**Comment by sunshinejr** [comment]
Hey guys, what is needed for the router to be merged in? :)

**Comment by Ben-G** [comment]
@sunshinejr @rpassis Just a friendly reminder 😆 
Sorry, haven't been on top of my GitHub notifications lately!

**Comment by sunshinejr** [comment]
No worries, thank you! 🎉

**Pull request merged**